codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res

```diff
@@ -110,19 +110,36 @@ module Make = (
     }
   }
 
+  let wildcardTopicSelection = T.contracts->Belt.Array.flatMap(contract => {
+    contract.events->Belt.Array.keepMap(event => {
+      let module(Event) = event
+      let {isWildcard, topicSelections} =
+        Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+      isWildcard ? Some(LogSelection.make(~addresses=[], ~topicSelections)) : None
+    })
+  })
+
   let getLogSelectionOrThrow = (~contractAddressMapping): array<LogSelection.t> => {
-    T.contracts->Belt.Array.keepMap((contract): option<LogSelection.t> => {
+    T.contracts
+    ->Belt.Array.keepMap((contract): option<LogSelection.t> => {
       switch contractAddressMapping->ContractAddressingMap.getAddressesFromContractName(
         ~contractName=contract.name,
       ) {
       | [] => None
       | addresses =>
-        let topicSelection =
-          LogSelection.makeTopicSelection(~topic0=contract.sighashes)->Utils.unwrapResultExn
-
-        Some(LogSelection.make(~addresses, ~topicSelections=[topicSelection]))
+        switch contract.events->Belt.Array.flatMap(event => {
+          let module(Event) = event
+          let {isWildcard, topicSelections} =
+            Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+
+          isWildcard ? [] : topicSelections
+        }) {
+        | [] => None
+        | topicSelections => Some(LogSelection.make(~addresses, ~topicSelections))
+        }
       }
     })
+    ->Array.concat(wildcardTopicSelection)
```

Comment:
Non wildcard events need to generate their topic selections with the addresses passed in and wildcardTopicSelection is concatenated to these selections.